chrysom/listenerClientOptions.go

```diff
@@ -0,0 +1,179 @@
+// SPDX-FileCopyrightText: 2021 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package chrysom
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+var (
+	defaultValidateListenerOptions = ListenerOptions{
+		validateReader(),
+		validateGetListenerLogger(),
+		validateSetListenerLogger(),
+		validatePullInterval(),
+		validateListener(),
+	}
+	defaultListenerOptions = ListenerOptions{
+		PullInterval(defaultPullInterval),
+		GetListenerLogger(func(context.Context) *zap.Logger { return zap.NewNop() }),
+		SetListenerLogger(func(context.Context, *zap.Logger) context.Context { return context.Background() }),
+	}
+)
+
+// ListenerOption is a functional option type for ListenerClient.
+type ListenerOption interface {
+	apply(*ListenerClient) error
+}
+
+type ListenerOptions []ListenerOption
+
+func (opts ListenerOptions) apply(c *ListenerClient) (errs error) {
+	for _, o := range opts {
+		errs = errors.Join(errs, o.apply(c))
+	}
+
+	return errs
+}
+
+type optionFunc func(*ListenerClient) error
```

Comment:
Why is this just optionFunc  not listenerOptionFunc?

Reply:
I forgot to rename it before posting the pr